### ISSUE TITLE
Chaining cut regions gives inconsistent results

### ISSUE THREAD

**Issue by stonnes** [yt-project/yt#2233]
### Bug report

**Bug summary**

There are two ways to "chain" cut regions:
1) `a = ds.cut_region(selection)`
    `b = a.cut_region(next selection)`

or 

2) `b = ds.cut_region(selection & next selection)`

They do not give the same results, and I think 1) is giving the incorrect results.  This will be, I hope, a more clear update of issue #1093.

**Code for reproduction**

```python
import sys
import numpy as np
import yt 

ds = yt.load("/mnt/ceph/users/stonnesen/runs/IsolatedGalaxy/galaxy0030/galaxy0030")

r0 = ds.all_data()
r1 = ds.disk([0.5,0.5,0.5],[0,0,1],(4,"kpc"),(10,"kpc"))
r1c = r1.cut_region(["(obj['cylindrical_r'].in_units('kpc') > 2.0)"])
r2c = r1.cut_region(["(obj['density'] > 1e-26)"])
r3c = r1c.cut_region(["(obj['density'] > 1e-26)"])
r4c = r1.cut_region(["(obj['cylindrical_r'].in_units('kpc') > 2.0) & (obj['density'] > 1e-26)"])

print ('Only density cut:')
print ('total cell volume:',r2c.quantities.total_quantity([("index","cell_volume")]))
print ('total cell mass:',r2c.quantities.total_quantity([("gas","cell_mass")]))
print ('extrema z:',r2c.quantities.extrema([("index","z")]))
print ('extrema y:',r2c.quantities.extrema([("index","y")]))
print ('extrema x:',r2c.quantities.extrema([("index","x")]))
print ('extrema cyl_r:',r2c.quantities.extrema([("index","cylindrical_r")]))
 
print ('Only spatial cut:')
print ('total cell volume:',r1c.quantities.total_quantity([("index","cell_volume")]))
print ('total cell mass:',r1c.quantities.total_quantity([("gas","cell_mass")]))
print ('extrema z:',r1c.quantities.extrema([("index","z")]))
print ('extrema y:',r1c.quantities.extrema([("index","y")]))
print ('extrema x:',r1c.quantities.extrema([("index","x")]))
print ('extrema cyl_r:',r1c.quantities.extrema([("index","cylindrical_r")]))
    
print ('Added density cut')
print ('total cell volume:',r3c.quantities.total_quantity([("index","cell_volume")]))
print ('total cell mass:',r3c.quantities.total_quantity([("gas","cell_mass")]))
print ('extrema z:',r3c.quantities.extrema([("index","z")]))
print ('extrema y:',r3c.quantities.extrema([("index","y")]))
print ('extrema x:',r3c.quantities.extrema([("index","x")]))
print ('extrema cyl_r:',r3c.quantities.extrema([("index","cylindrical_r")]))

print ('Spatial and density cut on same line')
print ('total cell volume:',r4c.quantities.total_quantity([("index","cell_volume")]))
print ('total cell mass:',r4c.quantities.total_quantity([("gas","cell_mass")]))
print ('extrema z:',r4c.quantities.extrema([("index","z")]))
print ('extrema y:',r4c.quantities.extrema([("index","y")]))
print ('extrema x:',r4c.quantities.extrema([("index","x")]))
print ('extrema cyl_r:',r4c.quantities.extrema([("index","cylindrical_r")]))
```
  

**Actual outcome**

chaining with two cut_region commands on separate lines gives bad mass and volume results:  here they are each a factor of ten higher than when I use a single command with the "&" symbol (and higher than when I just use the first spatial cut command).  The extrema for each position variable are the same, however.

**Expected outcome**

The 'total cell volume' and 'total cell mass' should be the same in the printouts after "Added density cut" and "Spatial and density cut on the same line"

**Comment by MatthewTurk** [comment]
@stonnes Hi Stephanie, I've reduced it a bit:

```python
import yt 

ds = yt.load("IsolatedGalaxy/galaxy0030/galaxy0030")
r0 = ds.all_data()
r1 = ds.disk([0.5,0.5,0.5],[0,0,1],(4,"kpc"),(10,"kpc"))
r1c = r1.cut_region(["(obj['cylindrical_r'].in_units('kpc') > 2.0)"])
r2c = r1.cut_region(["(obj['density'] > 1e-26)"])
r3c = r1c.cut_region(["(obj['density'] > 1e-26)"])
r4c = r1.cut_region(["(obj['cylindrical_r'].in_units('kpc') > 2.0) & (obj['density'] > 1e-26)"])
r5c = r1.cut_region(["(obj['cylindrical_r'].in_units('kpc') > 2.0)", "(obj['density'] > 1e-26)"])
regions = [r1c, r2c, r3c, r4c, r5c]

vs = [_.quantities.total_quantity([("index", "cell_volume"), ("gas","cell_mass")])
      for _ in regions ]

for (v, m), r in zip(vs, regions):
    print(v, r["ones"].shape)
```

This does indeed give different answers for everything after the first, and it shouldn't!

```
7.40736140869558e-07 code_length**3 (145172,)
2.953602233901621e-07 code_length**3 (123750,)
2.3316351871471856e-06 code_length**3 (96872,)
2.3316351871471857e-07 code_length**3 (96872,)
2.3316351871471857e-07 code_length**3 (96872,)
```

**Comment by MatthewTurk** [comment]
Just to make it explicit, r3c, r4c and r5c should all have the same value -- and not just the same cell size.

**Comment by MatthewTurk** [comment]
More info:

```
>>> (r3c["cell_volume"] == r4c["cell_volume"]).all()
True
>>> r3c.sum("cell_volume") == r4c.sum("cell_volume")
array(False)
>>> r3c["cell_volume"].sum() == r4c["cell_volume"].sum()
array(True)
```

So it looks like it's in the quantities somewhere.
r3c.sum("cell_volume") == r4c.sum("cell_volume")

**Comment by MatthewTurk** [comment]
@brittonsmith and I dug in and it looks like the problem is that `chunks` is redefined in `CutRegion`.  Doing it this way doesn't allow for conditionals to be applied properly to the underlying chunking.

Two immediately obvious solutions:

 1. Disallow `cut_region` on `YTCutRegion` objects (sad)
 2. Instead of cutting and cut region, create a new one with the same base object that appends the new `conditionals` onto the existing (list) `conditionals`.

**Comment by stonnes** [comment]
So, I am not sure I get all the terminology--basically stop pretending we can do step-by-step cut regions in multiple lines/calls, and only let cut_regions have multiple requirements (I think "conditionals") with the "&" or "," within a single call?

If so, sounds great and the docs just need to be changed.  Can yt throw an error when people try and chain the impossible way?

**Comment by brittonsmith** [comment]
@stonnes, I think we can fix this by changing how cut_regions made from other cut_regions work. Matt and I discussed a fix that I have working now. I'm working on adding some new tests and a PR should come shortly.

**Comment by brittonsmith** [comment]
Resolved in PR #2234.
